Library/Homebrew/extend/os/linux/extend/pathname.rb

```diff
@@ -0,0 +1,21 @@
+class Pathname
+  # @private
+  def elf?
+    # List of file signatures: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+    read(4) == "\x7fELF"
+  end
+
+  # @private
+  def dynamic_elf?
+    if !which("readelf").nil?
+      match = popen_read("readelf", "-l", to_path)[/ DYNAMIC /]
+      return false if match.empty?
```

Comment:
I believe this suffers from `NoMethodError: undefined method `empty?' for nil:NilClass` when it doesn't match.
I suggest…
```ruby
return popen_read("readelf", "-l", to_path).include?(" DYNAMIC ")
```
Sorry this contradicts my previous advice.

Reply:
done